src/theme/book.js

```diff
@@ -1,620 +1,723 @@
+    e.clearSelection();
+    showTooltip(e.trigger, "Copied!");
+  });
 
-    clipboardSnippets.on('error', function (e) {
-        showTooltip(e.trigger, "Clipboard error!");
-    });
+  clipboardSnippets.on("error", function(e) {
+    showTooltip(e.trigger, "Clipboard error!");
+  });
 })();
 
-(function scrollToTop () {
-    var menuTitle = document.querySelector('.menu-title');
+(function scrollToTop() {
+  var menuTitle = document.querySelector(".menu-title");
 
-    menuTitle.addEventListener('click', function () {
-        document.scrollingElement.scrollTo({ top: 0, behavior: 'smooth' });
-    });
+  menuTitle.addEventListener("click", function() {
+    document.scrollingElement.scrollTo({ top: 0, behavior: "smooth" });
+  });
 })();
 
 (function autoHideMenu() {
-    var menu = document.getElementById('menu-bar');
-
-    var previousScrollTop = document.scrollingElement.scrollTop;
-
-    document.addEventListener('scroll', function () {
-        if (menu.classList.contains('folded') && document.scrollingElement.scrollTop < previousScrollTop) {
-            menu.classList.remove('folded');
-        } else if (!menu.classList.contains('folded') && document.scrollingElement.scrollTop > previousScrollTop) {
-            menu.classList.add('folded');
-        }
-
-        if (!menu.classList.contains('bordered') && document.scrollingElement.scrollTop > 0) {
-            menu.classList.add('bordered');
-        }
+  var menu = document.getElementById("menu-bar");
+
+  var previousScrollTop = document.scrollingElement.scrollTop;
+
+  document.addEventListener(
+    "scroll",
+    function() {
+      if (
+        menu.classList.contains("folded") &&
+        document.scrollingElement.scrollTop < previousScrollTop
+      ) {
+        menu.classList.remove("folded");
+      } else if (
+        !menu.classList.contains("folded") &&
+        document.scrollingElement.scrollTop > previousScrollTop
+      ) {
+        menu.classList.add("folded");
+      }
+
+      if (
+        !menu.classList.contains("bordered") &&
+        document.scrollingElement.scrollTop > 0
+      ) {
+        menu.classList.add("bordered");
+      }
+
+      if (
+        menu.classList.contains("bordered") &&
+        document.scrollingElement.scrollTop === 0
+      ) {
+        menu.classList.remove("bordered");
+      }
+
+      previousScrollTop = document.scrollingElement.scrollTop;
+    },
+    { passive: true }
+  );
+})();
 
-        if (menu.classList.contains('bordered') && document.scrollingElement.scrollTop === 0) {
-            menu.classList.remove('bordered');
-        }
+(function serviceWorker() {
+  var isLocalhost =
+    ["localhost", "127.0.0.1", ""].indexOf(document.location.hostname) !== -1;
 
-        previousScrollTop = document.scrollingElement.scrollTop;
-    }, { passive: true });
+  if ("serviceWorker" in navigator && !isLocalhost) {
+    navigator.serviceWorker.register("sw.js").catch(function(error) {
```

Comment:
Encountered this error when navigate to pages other than base URL. 

```
Service worker registration failed: TypeError: "ServiceWorker script 
at http://localhost:3000/format/sw.js 
for scope http://localhost:3000/format/ encountered an error during installation."
```

The problem here is that the register path is relative to current file URL. We can resolve it by adding `<base href="https://our.basedomain.com">` though this would be a huge change.

Reply:
I feel that we could merge this PR and open another if the problem is too thorny to deal with at this time.